src/OhmmsSoA/readme.md

```diff
@@ -0,0 +1,39 @@
+ParticleAttribSoA
```

Comment:
For now, and the likely future, we are putting developer notes in the main manual to make them more accessible and less likely to be forgotten about.